controllers/machine_controller.go

```diff
@@ -440,6 +441,11 @@ func (r *MachineReconciler) isDeleteNodeAllowed(ctx context.Context, cluster *cl
 			if util.IsExternalManagedControlPlane(controlPlane) {
 				return nil
 			}
+
+			// Return early if the object referenced by controlPlaneRef is being deleted.
```

Comment:
I think we should move this check before the check for external managed control plane. An externally-managed control plane is also a valid control plane reference, and it can have a deletion timestamp too. Shouldn't we skip draining if an externally-managed control plane is being deleted, to be consistent for all control planes being deleted?

Reply:
yes. you're right. I had some misunderstanding about the check. updated